server/src/main/java/io/deephaven/server/table/ops/GrpcTableOperation.java

```diff
@@ -13,66 +14,81 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.function.Function;
+import java.util.stream.Collectors;
 
 public abstract class GrpcTableOperation<T> {
     @FunctionalInterface
     protected interface MultiDependencyFunction<T> {
-        List<TableReference> apply(T request);
+        List<TableReference> getTableReferences(T request);
+    }
+    @FunctionalInterface
+    protected interface PermissionFunction<T> {
+        void check(AuthContext authContext, T request, List<Table> sourceTables);
     }
 
+    final PermissionFunction<T> permission;
```

Comment:
I notice that the ops are referencing `permission` directly - that's not how the other member variables are used here (I would look into making them all private, and expose protected method for callers can call into to match other usage patterns). The nice thing is that you could have `protected void check(T request, List<SessionState.ExportObject<Table>> sourceTables)` and the caller wouldn't need to be responsible for providing the auth context or doing the `toTables`.

Reply:
The other approach would be to remove `PermissionFunction` entirely, and expose the member variable `TableServiceContextualAuthWiring` and the callers could call `auth.myAppropriateMethod(...)` directly (I'm not sure how useful abstracting PermissionFunction is...).